### PR TITLE
Add Kamikaze

### PR DESCRIPTION
**Pull request by Ultradragon005**
brb have to also add the other files

### CODE REVIEW

**Comment by Ultradragon005** [comment]
Please review

**Comment by LezaiYa** [comment]
Please wait for the next development before adding new roles.

**Comment by NikoCat233** [comment]
Also pls add your role in divatator results

File: Roles/Impostor/Kamikaze.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        
+        var KamiId = reader.ReadByte();
+        var targetId = reader.ReadByte();
+        byte KillerIdo = reader.ReadByte();
+        int Limit = reader.ReadInt32();
+        
+        KamikazedList[targetId] = KamiId;
+        MarkedLim.Add(KillerIdo, Limit);
+        MarkedLim[KillerIdo] = Limit;
```
Comment (Gurge44) [inline Roles/Impostor/Kamikaze.cs:76]: You can remove the .Add statement as the statement after it does the same but without exceptions

File: Roles/Impostor/Kamikaze.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        {
+            KamikazedList.Remove(kamikameha.PlayerId);
+        }
+        else if(CheckKamiDeath)
+        {
+            foreach (var alivePlayer in Main.AllAlivePlayerControls)
+            {
+                Main.PlayerStates[kamikameha.PlayerId].deathReason = PlayerState.DeathReason.Targeted;
+                alivePlayer.RpcMurderPlayerV3(kamikameha);
+            }
```
Comment (Gurge44) [inline Roles/Impostor/Kamikaze.cs:93]: Why do all alive players murder this one single player?!

File: Patches/PlayerControlPatch.cs [hunk 3]
```diff
@@ -2504,6 +2507,8 @@ public static void Postfix(PlayerControl __instance)
             LocateArrow.OnFixedUpdate(player);
         }
 
+        if (Kamikaze.IsEnable)
+            Kamikaze.MurderKamikazedPlayers(player);
```
Comment (Gurge44) [inline Patches/PlayerControlPatch.cs:2511]: As this isn't a priority, put this into`if (!lowLoad)` and this should only run on the host's side, so move it into `if (AmongUsClient.Instance.AmHost)`

File: Roles/Impostor/Kamikaze.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+using UnityEngine;
+using static TOHE.Options;
+using static TOHE.Translator;
+
+namespace TOHE.Roles.Impostor;
+
+public static class Kamikaze
+{
+    private static readonly int Id = 26900;
+    public static List<byte> MaxMarked = new();
```
Comment (Gurge44) [inline Roles/Impostor/Kamikaze.cs:15]: I couldn't find any use for this list, what is the purpose of it exactly?

File: Roles/Impostor/Kamikaze.cs (hunk 2 above)
Comment (Ultradragon005) [reply Roles/Impostor/Kamikaze.cs:97]: OH RIGHT, I forgot to check if aliveplayer corresponds with kamikameha player. Fixed now.

**Comment by 0xDrMoe** [comment]
Is this ready?

**Comment by Ultradragon005** [comment]
> Is this ready?

yup!

**Pull request merged**